source/client.ts

```diff
@@ -106,5 +106,11 @@ if (typeof preloaded === 'object') {
   }
 });
 
+const originalSetLocale = i18n.setLocale;
+i18n.setLocale = (locale, options, shouldVerifyLocale = false) =>
+shouldVerifyLocale && i18n.getLocale() === i18n.normalize(locale)
+    ? Promise.resolve()
+    : originalSetLocale.call(i18n, locale, options);
```

Comment:
1. It shouldn't be implemented as _patching_ the function, but rather implementing it in the `setLocale` directly.
2. We've decided to make this option _global_, not _per-call_. So instead of a third parameter here, we do have a new option in the options, configurable with `setOptions`.
    * I think that `ignoreNoopLocaleChanges` is a long and clear enough name.